types/rke_types.go

```diff
@@ -556,6 +556,9 @@ type CloudProvider struct {
 	OpenstackCloudProvider *OpenstackCloudProvider `yaml:"openstackCloudProvider,omitempty" json:"openstackCloudProvider,omitempty"`
 	// VsphereCloudProvider
 	VsphereCloudProvider *VsphereCloudProvider `yaml:"vsphereCloudProvider,omitempty" json:"vsphereCloudProvider,omitempty"`
+	// HarvesterCloudProvider is a string returned by calling the Harvester API "/v1/harvester/kubeconfig"
+	// We should synchronize this field to rancher
+	HarvesterCloudProvider string `yaml:"harvesterCloudProvider,omitempty" json:"harvesterCloudProvider,omitempty"`
```

Comment:
I think there should be related generated codes for this change?

Reply:
Yes, we should bump the rke package in Rancher.

Reply:
I think law is referring to https://github.com/rancher/rke/blob/master/types/zz_generated_deepcopy.go

Reply:
Rancher refers rke/types here.  https://github.com/rancher/rancher/blob/ea7823e463847ac1faf3dfdd62dbab7eec48bde1/pkg/apis/management.cattle.io/v3/cluster_types.go#L109

Reply:
There should be the related generated codes, but I can't generate code by the command `go generate`. I am still figuring out the reason.

Reply:
This is a string so it does not need a deepcopy method. @gitlawr @guangbochen

Reply:
Can this be a go struct so that it match the format with other cloud providers? Any reason to use string here? Also you get a benefit of adding more fields in harvester cloud provider later without having to tweak the api.